src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/pdf/PdfBoxService.java

```diff
@@ -0,0 +1,212 @@
+            String dmStoreDocumentName = generatePdfFileName(caseData, documentName);
+            return createDocumentInfoFromBytes(userToken, bytes, dmStoreDocumentName, caseTypeId);
+        } catch (IOException ioe) {
+            logException(String.format(UNABLE_TO_PROCESS_PDF_SOURCE, pdfTemplate),
+                    caseData.getEthosCaseReference(), ioe.getMessage(), PDF_SERVICE_CLASS_NAME,
+                    GENERATE_PDF_DOCUMENT_INFO_SERVICE_NAME);
+            throw ioe;
+        }
+    }
+
+    private static void throwPdfServiceExceptionWhenStringValueIsEmpty(String stringValue,
+                                                                       String exceptionMessage,
+                                                                       String firstWord,
+                                                                       String caseReferenceNumber)
+            throws PdfBoxServiceException {
+        if (StringUtils.isBlank(stringValue)) {
+            Throwable throwable = new Throwable(PDF_SERVICE_EXCEPTION_FIRST_WORD_WHEN_REQUIRED_FIELD_EMPTY);
+            throw new PdfBoxServiceException(exceptionMessage, throwable, firstWord, caseReferenceNumber,
+                    PDF_SERVICE_CLASS_NAME, GENERATE_PDF_DOCUMENT_INFO_SERVICE_NAME);
+        }
+    }
+
+    private static String generatePdfFileName(CaseData caseData, String documentName) {
+        return String.format(documentName, ObjectUtils.isEmpty(caseData.getSubmitEt3Respondent())
+                || StringUtils.isBlank(caseData.getSubmitEt3Respondent().getSelectedLabel())
+                ? PDF_OUTPUT_FILE_NAME_PREFIX : caseData.getSubmitEt3Respondent().getSelectedLabel());
+    }
+
+    private DocumentInfo createDocumentInfoFromBytes(String authToken, byte[] bytes, String documentName,
+                                                    String caseTypeId) {
+        URI documentSelfPath = uploadDocument(documentName, authToken, bytes, caseTypeId);
+        log.info(String.format(PDF_DOCUMENT_CREATED_LOG_INFO, documentSelfPath.toString()));
+        String downloadUrl = documentManagementService.generateDownloadableURL(documentSelfPath);
+        String markup = documentManagementService.generateMarkupDocument(downloadUrl);
+        return generateDocumentInfo(documentName, documentSelfPath, markup);
+    }
+
+    private URI uploadDocument(String documentName, String authToken, byte[] bytes, String caseTypeId) {
+        return documentManagementService.uploadDocument(authToken, bytes, documentName,
+                    APPLICATION_PDF_VALUE, caseTypeId);
+    }
+
+    private DocumentInfo generateDocumentInfo(String documentName, URI documentSelfPath, String markupURL) {
+        return DocumentInfo.builder()
+                .type(APPLICATION_PDF_VALUE)
+                .description(documentName)
+                .markUp(markupURL)
+                .url(ccdGatewayBaseUrl + documentSelfPath.getRawPath() + "/binary")
+                .build();
+    }
```

Comment:
Is there a reason why this is duplicated as opposed to using the code in TornadoService?